wagtail/models/i18n.py

```diff
@@ -183,27 +183,33 @@ class TranslatableMixin(models.Model):
 
     class Meta:
         abstract = True
-        unique_together = [("translation_key", "locale")]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["translation_key", "locale"],
+                name="unique_translation_key_locale_%(app_label)s_%(class)s",
+            )
+        ]
 
     @classmethod
     def check(cls, **kwargs):
         errors = super().check(**kwargs)
         is_translation_model = cls.get_translation_model() is cls
 
-        # Raise error if subclass has removed the unique_together constraint
+        unique_constraint_fields = ("translation_key", "locale")
+
+        # Raise error if subclass has removed constraints
         # No need to check this on multi-table-inheritance children though as it only needs to be applied to
         # the table that has the translation_key/locale fields
-        if (
-            is_translation_model
-            and ("translation_key", "locale") not in cls._meta.unique_together
-        ):
+        if is_translation_model and unique_constraint_fields in cls._meta.constraints:
             errors.append(
                 checks.Error(
```

Comment:
We'll need to update the tests as well. We currently have this:

https://github.com/wagtail/wagtail/blob/56c62cc29874b98c863792628fa5cdf40db1589b/wagtail/tests/test_translatablemixin.py#L190-L201

We need to have another one that ensures a model that uses `UniqueConstraint` (and not `unique_together` will not produce an error.

Reply:
Yeah, I updated this already locally. But would that still be necessary if we still want to retain both `unique_together` and `UniqueConstraint`. I was thinking it might not longer be since `unique_together` would still be in use. Perhaps, I can modify it to also check that the model now has a warning too.